csp_billing_adapter/bill_utils.py

```diff
@@ -113,9 +133,9 @@ def get_billable_usage(
 
     for metric, data in config.usage_metrics.items():
```

Comment:
In fact you could do the filtering on the usage_record list just before this line instead, to select the relevant ones to process, without needing to call it in both subroutines, or change the function signatures of the get_{average,maximum}_usage subroutines, and the helper function, while still useful (in that it could have a self-documenting name to indicate that it is filter application records) is not required.